Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +127,352 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+String valueToTrustedHTMLString(JSGlobalObject& lexicalGlobalObject, JSValue value, const String& sink, bool nullToEmpty)
+{
+    // TODO: check settings for feature flag in all these functions
+    VM& vm = lexicalGlobalObject.vm();
+
+    if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, value))
```

Comment:
TODO extract shared code into common functions to avoid so much code duplication. Probably best to follow the algorithm breakdown within the spec for optimal reusability.